### PR TITLE
runtime: clean up realtime impl namespace

### PR DESCRIPTION
**Pull request by mormj**
Renames the gr::impl namespace to gr::realtime and moves impl stuff inside lib/ 
Removes the impl stuff from the python bindings
Some other minor cleanup

Addresses #3028

### CODE REVIEW

**Comment by drmpeg** [comment]
I think that grc/tests/resources/top_block.py file is spurious. It gets created when you do ```make test```.

**Comment by mormj** [comment]
good catch!  need to get that into the .gitignore

File: gnuradio-runtime/include/gnuradio/realtime.h [hunk 1]
```diff
@@ -12,10 +12,16 @@
 #define INCLUDED_GR_REALTIME_H
 
 #include <gnuradio/api.h>
-#include <gnuradio/realtime_impl.h>
 
 namespace gr {
 
+typedef enum { RT_OK = 0, RT_NOT_IMPLEMENTED, RT_NO_PRIVS, RT_OTHER_ERROR } rt_status_t;
+
+enum rt_sched_policy {
```
Comment (ThomasHabets) [inline gnuradio-runtime/include/gnuradio/realtime.h:20]: Maybe switch to `enum class` while you're at it, for better type safety?

**Review by ThomasHabets — CHANGES_REQUESTED** [review]

File: gnuradio-runtime/include/gnuradio/realtime.h (hunk 1 above)
Comment (mbr0wn) [reply gnuradio-runtime/include/gnuradio/realtime.h:20]: I also prefer enum classes, but that's an API break that we might be able to avoid.

**Review by ThomasHabets — APPROVED** [review]

**Review by mbr0wn — APPROVED** [review]

**Pull request merged**